enum.c

```diff
@@ -2759,7 +2759,9 @@ each_with_object_i(RB_BLOCK_CALL_FUNC_ARGLIST(i, memo))
  *    enum.each_with_object(obj)                              ->  an_enumerator
  *
  *  Iterates the given block for each element with an arbitrary
- *  object given, and returns the initially given object.
+ *  object given, and returns the initially given object.  The 
+ *  object must not be reassigned within the block, or its value
+ *  in that iteration will be ignored.
```

Comment:
I don't think it makes sense to add this. I can kind of understand the point you are making with the example given in the PR, but the text by itself doesn't make sense. Objects are not reassigned, variables (including block and method arguments) are.  It's a general principle in Ruby that reassigning a variable doesn't modify the object previously referenced by the variable.  Certainly the behavior you are trying to warn against is not specific to the `each_with_object` method, it would apply to many methods.

As a brief demonstration of the problem with the documentation you are suggesting:

```ruby
["a", "b", "c"].each_with_object([]) do |x, acc|
  acc << "#{x}-ok"
  # block argument reassigned, but change
  # made in this iteration is not ignored
  acc = ["potatoes"]
end
```

Reply:
It's true that that the example is obviously wrong.  My mental implementation of the `each_with_object` function makes that plain:

```ruby
def my_each_with_object(enum, input)
  enum.each { |e| yield(e, input) }
  input
end
```

(There's just no way for the outer function to react to what might be happening in the scope of the block.)

Still, I feel like there's something to be desired from this part of the current text:

> Iterates the given block for each element with an arbitrary object given, and returns the initially given object.

I think "with an arbitrary object given" might be trying to say "you can give the input object whatever name you want in the context of the block"?  I don't understand the significance there.  What's tripping me up is the use of "object" twice in that passage, which (for me) mentally turned into "the arbitrary object is somehow independent of the initially given object".

Reply:
OK, I've updated the text using the documentation for `inject` as a starting point.